pkg/virt-handler/node-labeller/arch_labeller.go

```diff
@@ -81,3 +87,64 @@ func (defaultArchLabeller) supportsHostModel() bool {
 func (defaultArchLabeller) arch() string {
 	return runtime.GOARCH
 }
+
+// prepareLabels converts cpu models, features, hyperv features to map[string]string format
+// e.g. "cpu-feature.node.kubevirt.io/Penryn": "true"
+func (defaultArchLabeller) prepareLabels(nodeLabeller *NodeLabeller, node *v1.Node, cpuModels []string, cpuFeatures cpuFeatures, hostCpuModel hostCPUModel, obsoleteCPUsx86 map[string]bool) map[string]string {
```

Comment:
```suggestion
func (nodeLabeller *defaultArchLabeller) prepareLabels(node *v1.Node, cpuModels []string, cpuFeatures cpuFeatures, hostCpuModel hostCPUModel, obsoleteCPUsx86 map[string]bool) map[string]string {
```

Pardon my ignorance, something like this won't work?

Reply:
The `NodeLabeller` has a reference to the `archLabeller` (which is basically a configuration of what is supported on each arch), and also it holds a lot of data related to the labels it needs to handle, they are not the same object

i agree about the fact that it looks weird, passing a reference to the original `NodeLabeller` into a method which is reside in the `archLabeller` now, but what alternative is there, the original method needs access to the data of the `NodeLabeller` (machine-types, cpu vendor, host-model etc..), and moving the `prepareLabels` seems like a good engineering because we want to support specific stuff when the arch is different then the default.

Reply:
I see